### PR TITLE
fix(eslint-plugin-vue): [valid-v-bind-sync] Added Case: when valid is then valid .sync

### LINKED ISSUE #1298: "vue/valid-v-bind-sync" is not accounting for "is" attribute
**What rule do you want to change?**
`vue/valid-v-bind-sync` 

**Does this change cause the rule to produce more or fewer warnings?**
Fewer

**Please provide some example code that this change will affect:**
```
<table>
    <tr is="my-row" 
        :some-prop.sync="somePropValue"
        :some-other-prop.sync="someOtherPropValue">
        <td></td>
    </tr>
</table>
```

**What does the rule currently do for this code?**
Displays the error/warning, `'.sync' modifiers aren't supported on <tr> non Vue-components`, for every prop containing the `.sync` modifier.

**What will the rule do after it's changed?**
Evaluate any `.sync` modifier implementation as valid if the target element has a valid `is` attribute.

**Additional context**
The `is` attribute is used for [dynamic components](https://vuejs.org/v2/guide/components.html#Dynamic-Components) and to work around [limitations of in-DOM templates](https://vuejs.org/v2/guide/components.html#DOM-Template-Parsing-Caveats) (see Vue [documentation](https://vuejs.org/v2/api/#is)):
```
<!-- necessary because `<my-row>` would be invalid inside -->
<!-- a `<table>` element and so would be hoisted out      -->
<table>
  <tr is="my-row"></tr>
</table>
```

### PR DESCRIPTION
**Pull request by aggmoulik**
Fixes #1298 
Signed-off-by: Moulik Aggarwal <qwertymoulik@gmail.com>

### CODE REVIEW

File: lib/rules/valid-v-bind-sync.js [hunk 1]
```diff
@@ -82,6 +82,20 @@ function maybeNullObjectMemberExpression(node) {
...
 }
 
+function isValidIs(node) {
+  const { attributes } = node
+  const isAttribute = attributes.filter((attr) => {
+    if (attr.type === 'VAttribute' && attr.key.name === 'is') return attr
+  })
+  if (
+    isAttribute.length > 0 &&
+    ['tr', 'li', 'option'].includes(node.parent.name)
```
Comment (ota-meshi) [inline lib/rules/valid-v-bind-sync.js:92]: Could you remove the element name check?
The `is` attribute can be used on all elements. There is no need to check the element name.

File: lib/rules/valid-v-bind-sync.js [hunk 2]
```diff
@@ -82,6 +82,20 @@ function maybeNullObjectMemberExpression(node) {
   return false
 }
 
+function isValidIs(node) {
+  const { attributes } = node
+  const isAttribute = attributes.filter((attr) => {
+    if (attr.type === 'VAttribute' && attr.key.name === 'is') return attr
```
Comment (ota-meshi) [inline lib/rules/valid-v-bind-sync.js:88]: We should also check `v-bind:is`.

File: lib/rules/valid-v-bind-sync.js [hunk 3]
```diff
@@ -82,6 +82,20 @@ function maybeNullObjectMemberExpression(node) {
   return false
 }
 
+function isValidIs(node) {
+  const { attributes } = node
+  const isAttribute = attributes.filter((attr) => {
```
Comment (ota-meshi) [inline lib/rules/valid-v-bind-sync.js:87]: Could you use `some` instead of `filter`.

**Review by ota-meshi — CHANGES_REQUESTED** [review]
Thank you for this PR!
I have some change requests.

**Comment by aggmoulik** [comment]
Hy @ota-meshi I have updated your changes and added a cli utility also, as it helps us to check individual rules and makes our development faster.

**Comment by aggmoulik** [comment]
Also, Add `hacktoberfest` label in the PR or in the repository tags.

**Comment by ota-meshi** [comment]
I'm currently maintaining this repository almost alone, so I can't afford to accept all PR during the `hacktoberfest`, so I won't add `hacktoberfest` label.
Also, I'm sorry, this PR can't promise to complete all reviews and merge in October.

**Comment by aggmoulik** [comment]
@ota-meshi I think I can help you in that as I am working to be a active-contributor not for just hacktoberfest. I can help in code review as I already do in my company.

File: lib/rules/valid-v-bind-sync.js [hunk 4]
```diff
@@ -82,6 +82,22 @@ function maybeNullObjectMemberExpression(node) {
...
+function isValidIs(node) {
+  const { attributes } = node
+  const isAttribute = attributes.some((attr) => {
+    // check for `VAttribute`
+    if (attr.type === 'VAttribute') {
+      // check for `is` attribute
+      if (attr.key.type === 'VIdentifier' && attr.key.name === 'is') return true
+
+      // check for `:is` `bind` attribute
+      if (attr.key.type === 'VDirectiveKey' && attr.key.argument.name === 'is')
```
Comment (ota-meshi) [inline lib/rules/valid-v-bind-sync.js:94]: You need to check if it is v-bind. Also, the argument can be null.

https://github.com/vuejs/vue-eslint-parser/blob/master/docs/ast.md#vdirectivekey

In addition, could you add a test to make sure it is not judged as `is`?

```vue
<tr v-on:is="myRow" :some-prop.sync="somePropValue">
<tr v-bind="myRow" :some-prop.sync="somePropValue">
```

File: package.json [hunk 5]
```diff
@@ -19,7 +19,8 @@
     "version": "npm run lint -- --fix && git add .",
     "update": "node ./tools/update.js",
     "docs:watch": "vuepress dev docs",
-    "docs:build": "vuepress build docs"
+    "docs:build": "vuepress build docs",
+    "tests": "mocha"
```
Comment (ota-meshi) [inline package.json:23]: Do not change the script. You may be able to use `npx mocha`.

File: tests/lib/rules/valid-v-bind-sync.js [hunk 6]
```diff
@@ -145,6 +145,48 @@ tester.run('valid-v-bind-sync', rule, {
...
+        </table>
+      </template>
+      `
+    },
+    {
+      filename: 'test.vue',
+      code: `
+      <template>
+        <table>
+          <tr v-bind:is="my-row"
```
Comment (ota-meshi) [inline tests/lib/rules/valid-v-bind-sync.js:182]:
```suggestion
          <tr v-bind:is="myRow"
```

**Review by ota-meshi — CHANGES_REQUESTED** [review]
Sorry for the late reply. I posted some comments.

File: package.json (hunk 5 above)
Comment (ota-meshi) [inline package.json:23]:
```suggestion
    "docs:build": "vuepress build docs"
```

**Comment by ota-meshi** [comment]
I would like to share the fix related to this issue with another fix. So I'll fix this PR and merge it.

**Pull request merged**